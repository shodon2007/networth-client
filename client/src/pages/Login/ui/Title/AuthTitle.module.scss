.MainBoldFont{
  font-family: var(--font-main-Blin);
  font-weight: 600;
  font-size: 64px;
  color: #d9d9d9;
}