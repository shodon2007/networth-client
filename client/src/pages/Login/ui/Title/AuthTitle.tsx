import { ReactNode } from "react"
import cls from './AuthTitle.module.scss'

interface AuthTitleProps { 
  children: ReactNode
}

export const AuthTitle = ({children}: AuthTitleProps) => {
  return (
    <h1 className={cls.MainBoldFont}>{children}</h1>
  )
}