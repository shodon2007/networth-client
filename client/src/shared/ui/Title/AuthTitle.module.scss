.MainBoldFont{
  font-family: var(--font-main-Blin);
  font-weight: 600;
  font-size: 64px;
  color: var(--white-color);
}