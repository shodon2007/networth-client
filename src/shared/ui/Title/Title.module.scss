.title,
.middle {
	font-family: var(--font-main-Blin);
	font-weight: 600;
	font-size: 42px;
	color: var(--white-color);
}

.small {
	font-size: 26px;
}

.big {
	font-size: 64px;
}
