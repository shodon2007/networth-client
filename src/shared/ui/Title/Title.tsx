import {ReactNode} from "react";
import classNames from "src/shared/lib/classNames";

import cls from "./Title.module.scss";

export enum TitleType {
	small = "SMALL",
	middle = "MIDDLE",
	big = "BIG",
}

interface TitleProps {
	children: ReactNode;
	type?: TitleType;
}

export const Title = ({children, type = TitleType.middle}: TitleProps) => {
	return <h1 className={classNames(cls.title, {}, [cls[type]])}>{children}</h1>;
};
