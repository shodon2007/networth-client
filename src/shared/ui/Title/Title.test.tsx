import {render} from "@testing-library/react";
import {Title, TitleType} from "./Title";

describe("Testing Title.test.tsx", () => {
	test("testing default styles", () => {
		const screen = render(<Title>hello</Title>);
		const title = screen.getByText("hello");
		expect(title).toBeInTheDocument();
		expect(title).toMatchSnapshot();
	});
	describe("Testing types", () => {
		test("testing middle type", () => {
			const screen = render(<Title type={TitleType.MIDDLE}>hello</Title>);
			const title = screen.getByText("hello");
			expect(title).toBeInTheDocument();
			expect(title).toMatchSnapshot();
		});
		test("testing small type", () => {
			const screen = render(<Title type={TitleType.SMALL}>hello</Title>);
			const title = screen.getByText("hello");
			expect(title).toBeInTheDocument();
			expect(title).toMatchSnapshot();
		});
		test("testing big type", () => {
			const screen = render(<Title type={TitleType.BIG}>hello</Title>);
			const title = screen.getByText("hello");
			expect(title).toBeInTheDocument();
			expect(title).toMatchSnapshot();
		});
	});
});
