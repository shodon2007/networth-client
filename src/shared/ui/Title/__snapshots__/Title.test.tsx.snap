// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Title.test.tsx Testing types testing big type 1`] = `
<h1
  class="title BIG"
>
  hello
</h1>
`;

exports[`Testing Title.test.tsx Testing types testing middle type 1`] = `
<h1
  class="title MIDDLE"
>
  hello
</h1>
`;

exports[`Testing Title.test.tsx Testing types testing small type 1`] = `
<h1
  class="title SMALL"
>
  hello
</h1>
`;

exports[`Testing Title.test.tsx testing default styles 1`] = `
<h1
  class="title MIDDLE"
>
  hello
</h1>
`;
